gemstone/generator/port_reference.py

```diff
@@ -101,3 +102,15 @@ def qualified_name(self):
         for op in self._ops:
             name = op.transform_name(name)
         return name
+
+    @staticmethod
+    def __get_bit_width(t):
```

Comment:
This is not robust, because we have other types like tuple and array, and nested types. However, because all of our base types are `Bit` we can do some nested hash on the magma type.

Reply:
I'd prefer have magma types come with their own hash function. Until then I'd prefer to leave it as is.